sklearn/decomposition/tests/test_pca.py

```diff
@@ -508,3 +509,15 @@ def fit_deprecated(X):
     assert_warns_message(DeprecationWarning, depr_message, fit_deprecated, X)
     Y_pca = PCA(svd_solver='randomized', random_state=0).fit_transform(X)
     assert_array_almost_equal(Y, Y_pca)
+
+
+def test_pca_spase_input():
+
+    X = np.random.rand(5, 4)
```

Comment:
You need to use a seeded RNG, for instance "X = np.random.RandomState(0).rand(5, 4)". We really don't want to leave any non seeded RNG in the test suite.